### ISSUE TITLE
2.0.5 - MongoDB - Multiple mappings in object results in NullPointerException when persisted

### ISSUE THREAD

**Issue by bthor** [Impetus/kundera#43]
When an object has multiple mappings to other objects, the call to find the person object will result in a 
java.lang.NullPointerException
    at com.impetus.kundera.persistence.AbstractEntityReader.computeGraph(AbstractEntityReader.java:115)
    at com.impetus.kundera.persistence.PersistenceDelegator.find(PersistenceDelegator.java:755)
    at com.impetus.kundera.persistence.EntityManagerImpl.find(EntityManagerImpl.java:106)

This is because the persisted object will not have Person.blaId stored in mongo.  I'm guessing the call to write Address over writes the blaId?

Here is the complete code (3 objects and a test driver):

@Entity
@Table( schema = "test@testMongo", name="Bla")
public class Bla {

```
@Id
@Column(name="blaId")
private String blaId;

@Column(name = "bla")
private String bla;

@OneToOne(mappedBy="bla")
private Person person;

public String getBlaId() {
    return blaId;
}

public void setBlaId(String blaId) {
    this.blaId = blaId;
}

public String getBla() {
    return bla;
}

public void setBla(String bla) {
    this.bla = bla;
}

public Person getPerson() {
    return person;
}
public void setPerson(Person person) {
    this.person = person;
}
```

}

@Entity
@Table( schema = "test@testMongo", name="Person")
public class Person {

```
@Id
@Column(name="personId")
private String personId;

@OneToOne(cascade=CascadeType.ALL, fetch=FetchType.LAZY)
@JoinColumn(name="blaId")
private Bla bla;

@ManyToOne(cascade=CascadeType.ALL, fetch=FetchType.LAZY)
@JoinColumn(name="addressId")
private Address address;

public String getPersonId() {
    return personId;
}

public void setPersonId(String personId) {
    this.personId = personId;
}

public Address getAddress() {
    return address;
}

public void setAddress(Address address) {
    this.address = address;
}

public Bla getBla() {
    return bla;
}

public void setBla(Bla bla) {
    this.bla = bla;
}
```

}

@Entity
@Table( schema = "test@testMongo", name="Address")
public class Address {

```
@Id
@Column(name="addressId")
private String addressId;

@Column(name = "street")
private String street;

@OneToMany(mappedBy= "address", fetch = FetchType.LAZY)
private List<Person> persons;


public String getAddressId() {
    return addressId;
}

public void setAddressId(String addressId) {
    this.addressId = addressId;
}

public String getStreet() {
    return street;
}

public void setStreet(String street) {
    this.street = street;
}
```

}

public class Test {
    public static void main(String[] args) {

```
    EntityManagerFactory emf = Persistence.createEntityManagerFactory("testMongo");
    EntityManager em = emf.createEntityManager();

    Address address = new Address();
    address.setAddressId(UUID.randomUUID().toString());
    address.setStreet("123 Kundera St.");

    Bla bla = new Bla();
    bla.setBlaId(UUID.randomUUID().toString());
    bla.setBla("case of the blas");

    String id = UUID.randomUUID().toString();
    Person person = new Person();
    person.setPersonId(id);
    person.setAddress(address);
    person.setBla(bla);

    em.persist(person);
    person = em.find(Person.class, id); // this throws the NPE

    em.close();
    emf.close();
}
```

}

**Comment by mevivs** [comment]
I am looking into this.

**Comment by mevivs** [comment]
issue is reproducible and will be fixed in 2.0.6

**Comment by kkmishra** [comment]
verified, working fine with 2.0.6
